flake.nix

```
@@ -1,5 +1,33 @@
 {
   description = "Allow flakes to be used with Nix < 2.4";
 
-  outputs = _: { };
+  outputs =
+    { self, ... }:
+    let
+      allOutputs = publicOutputs // {
+        # Use explicit inherit, to ensure allOutputs is evaluated without
+        # evaluating devOutputs.
+        inherit (devOutputs) devShells checks;
```

let's add `formatter` so one can run `nix fmt` 

```suggestion
        inherit (devOutputs) devShells checks formatter;
```

---

~~Maybe we could also add treefmt to devShell? I find it useful when starting my editor from project's shell to have local treefmt picked up for formatting on save~~

EDIT: it's there already